### PR TITLE
decompile unk_020023C0.s into render_text.c

### PR DESCRIPTION
**Pull request by remicalixte**
Most functions are similar to ones in pokeemerald https://github.com/pret/pokeemerald/blob/master/gflib/text.c

### CODE REVIEW

File: arm9/src/render_text.c [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+THUMB_FUNC u32 RenderText(struct TextPrinter *printer)
+{
+
+    struct TextPrinterSubStruct *subStruct =
+        (struct TextPrinterSubStruct *)(&printer->subStructFields);
+    u16 currentChar;
+
+    switch (printer->state)
+    {
+    case 0:
```
Comment (red031000) [inline arm9/src/render_text.c:26]: indentation here

File: arm9/src/render_text.c [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+                printer->printerTemplate.lineSpacing + fontAttribute;
+
+            return 2;
+
+        case 0xF0FD:
+            printer->printerTemplate.currentChar.raw++;
+            return 2;
+        case 0xFFFE:
+            printer->printerTemplate.currentChar.raw--;
+            switch ((u16)MsgArray_GetControlCode(printer->printerTemplate.currentChar.raw))
```
Comment (red031000) [inline arm9/src/render_text.c:77]: indentation

File: include/render_text.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#include "global.h"
```
Comment (red031000) [inline include/render_text.h:1]: header guards

**Review by red031000 — CHANGES_REQUESTED** [review]
few things

File: arm9/src/render_text.c (hunk 1 above)
Comment (remicalixte) [reply arm9/src/render_text.c:26]: How do you want the indentation to be? It's set this way by clang-format

Comment (red031000) [reply arm9/src/render_text.c:26]: cases etc should be one in from the { imo, it's fine as is though I guess

Comment (remicalixte) [reply arm9/src/render_text.c:26]: It's specifically set to false here: https://github.com/pret/pokediamond/blob/54b6645c02958cff375393a5261d5376dd169b94/.clang-format#L11

But I can change that if you want

**Review by ProjectRevoTPP — DISMISSED** [review]

File: arm9/src/render_text.c (hunk 1 above)
Comment (red031000) [reply arm9/src/render_text.c:26]: might be a good idea

Comment (remicalixte) [reply arm9/src/render_text.c]: done

**Review by red031000 — APPROVED** [review]

**Pull request merged**